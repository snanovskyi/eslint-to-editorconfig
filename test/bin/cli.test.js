const fs = require('fs');
const path = require('path');
const execa = require('execa');

describe('cli', () => {
  const { cwd } = process;
  let cli;
  const eslintConfigPath = path.join(__dirname, '.eslintrc.json');
  const eslintConfig = {
    rules: {
      indent: ['error', 2],
      'no-trailing-spaces': 'warn',
      'eol-last': 'error',
      'max-len': ['warn', 80],
    },
  };
  const editorconfigPath = path.join(__dirname, '.editorconfig');

  beforeAll(() => {
    process.cwd = () => path.join(__dirname, '..', '..');
    cli = path.join(process.cwd(), 'bin', 'cli.js');
  });

  afterAll(() => {
    process.cwd = cwd;
  });

  beforeEach(() => {
    fs.writeFileSync(eslintConfigPath, JSON.stringify(eslintConfig));
  });

  afterEach(() => {
    fs.unlinkSync(eslintConfigPath);
    if (fs.existsSync(editorconfigPath)) {
      fs.unlinkSync(editorconfigPath);
    }
  });

  it("should create new '.editorconfig' file if does not exist", () => {
    execa.sync(cli, [eslintConfigPath, '--output', editorconfigPath]);
    const editorconfig = fs.readFileSync(editorconfigPath).toString();
    expect(editorconfig).toMatchSnapshot();
  });

  it("should append rules if '.editorconfig' already exists", () => {
    fs.writeFileSync(editorconfigPath, 'root = true');
    execa.sync(cli, [eslintConfigPath, '--output', editorconfigPath]);
    const editorconfig = fs.readFileSync(editorconfigPath).toString();
    expect(editorconfig).toMatchSnapshot();
  });

  it("should handle 'cwd' with absolute 'config' and 'output'", () => {
    execa.sync(cli, [eslintConfigPath, '--output', editorconfigPath, '--cwd', process.cwd()]);
    const editorconfig = fs.readFileSync(editorconfigPath).toString();
    expect(editorconfig).toMatchSnapshot();
  });

  it('should handle "stdout" option', () => {
    const { stdout } = execa.sync(cli, [
      eslintConfigPath,
      '--output',
      editorconfigPath,
      '--stdout',
    ]);
    const editorconfigExists = fs.existsSync(editorconfigPath);
    expect(editorconfigExists).toBeFalsy();
    expect(stdout).toMatchSnapshot();
  });
});
