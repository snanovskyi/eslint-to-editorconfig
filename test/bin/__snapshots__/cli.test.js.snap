// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cli should append rules if '.editorconfig' already exists 1`] = `
"root = true
[*.js]
indent_style = space
indent_size = 2
trim_trailing_whitespace = true
insert_final_newline = true
max_line_length = 80"
`;

exports[`cli should create new '.editorconfig' file if does not exist 1`] = `
"[*.js]
indent_style = space
indent_size = 2
trim_trailing_whitespace = true
insert_final_newline = true
max_line_length = 80"
`;

exports[`cli should handle "stdout" option 1`] = `
"indent_style = space
indent_size = 2
trim_trailing_whitespace = true
insert_final_newline = true
max_line_length = 80"
`;

exports[`cli should handle 'cwd' with absolute 'config' and 'output' 1`] = `
"[*.js]
indent_style = space
indent_size = 2
trim_trailing_whitespace = true
insert_final_newline = true
max_line_length = 80"
`;
